azure-storage-queue/src/main/scala/akka/stream/alpakka/azure/storagequeue/scaladsl/AzureQueueSource.scala

```diff
@@ -14,8 +14,8 @@ object AzureQueueSource {
    * Scala API: creates a [[AzureQueueSource]] for a Azure CloudQueue.
    */
   def apply(
-      cloudQueue: CloudQueue,
+      cloudQueue: () => CloudQueue,
       settings: AzureQueueSourceSettings = AzureQueueSourceSettings.default
   ): Source[CloudQueueMessage, NotUsed] =
-    Source.fromGraph(new AzureQueueSourceStage(cloudQueue, settings))
+    Source.fromGraph(new AzureQueueSourceStage(cloudQueue(), settings))
```

Comment:
As I said before, the Stage can be created once but materialized any number of times concurrently. This means the factory must be passed to the stage, and called to create a queue client when the `GraphStageLogic` is created to be safe.